### PR TITLE
Add tutorial how to create Prometheus exporter Flow components

### LINKED ISSUE #3228: Add tutorial on how to create Flow `prometheus.exporter` components
Add tutorial that helps contributors to create Prometheus exporter Flow component from  Agent integrations.
Some things to consider and best practices.

- What are the different bits to be defined in a component that wraps an integration.
- Usage of blocks (when to use them, how to define them, field name in singular, etc).
- What are arguments and exports.
- Importance of unit testing.
- How to write component's documentation.
- Usage of `Secret` type when defining sensitive secrets.

Please, feel free to add more suggestions here that you consider that need to be covered in a tutorial likes this.

### PR DESCRIPTION
**Pull request by marctc**
#### PR Description

Add tutorial of best practices and tips how to write Flow components for Prometheus
exporters. This will help us to get adoption of more exporters in Flow. 

Ideally we should merge #3145 before this get merged (as it references it).

#### Which issue(s) this PR fixes

Fixes #3228

#### PR Checklist

- [ ] CHANGELOG updated
- [x] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: docs/developer/writing-exporter-flow-components.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	target "example" {
+		address = "http://example.com"
+		module  = "http_2xx"
+	}
+}
+```
+
+- Define `DefaultArguments` as a global variable to define the default arguments for the component. 
+
+## Functions
```
Comment (rfratto) [inline docs/developer/writing-exporter-flow-components.md:45]: What do you think about also documenting:

* The init function to register the component
* Updating components/all with the new component

File: docs/developer/writing-exporter-flow-components.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# Create Prometheus Exporter Flow Components
```
[the remark this answers is not in the document]
Comment (marctc) [reply docs/developer/writing-exporter-flow-components.md:1]: I linked process exporter and blackbox exporter somewhere in the middle of the docs, is that enough of you have something else in mind?

File: docs/developer/writing-exporter-flow-components.md (hunk 1 above)
Comment (marctc) [reply docs/developer/writing-exporter-flow-components.md:44]: Done for `init`. I'm not sure what you mean by the second point, could you clarify?

File: docs/developer/writing-exporter-flow-components.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+This guide will walk you through the process of creating a new Prometheus exporter Flow component and best practices for implementing it. It is required that the exporter has an existing integration in order to wrap it as a Flow component.
```
[the remark this answers is not in the document]
Comment (marctc) [reply docs/developer/writing-exporter-flow-components.md:3]: I can add a not saying that.

File: docs/developer/writing-exporter-flow-components.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+## Arguments (Configuration)
+
+`Arguments` struct defines the arguments that can be passed to the component. In most cases, this would be exactly the same as the arguments that the integration for this exporter the uses. Some recommendations:
+
+- Use `attr` tag to define the name of the argument in the Flow component.
+- Use `optional` tag for optional arguments.
+- Use `rivertypes.Secret` type for sensitive arguments (e.g. API keys, passwords, etc). The original integration should have a similar field type called `Secret` from Prometheus.
+- If one of the arguments is a slice, use `block`. For example, the [process_exporter](../../component/prometheus/exporter/process/process.go) `Arguments` struct has `ProcessExporter` param which is a `[]MatcherGroup`. The name of the parameter should be in singular. This will allow the user to define multiple blocks of the same type.
```
[the remark this answers is not in the document]
Comment (rfratto) [reply docs/developer/writing-exporter-flow-components.md]: > I wrote "The name of the parameter should be in singular", isn' that enough?

Oops, sorry, I totally missed that part. LGTM

File: docs/developer/writing-exporter-flow-components.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (rfratto) [reply docs/developer/writing-exporter-flow-components.md:1]: Ah, sorry, my eyes glazed over those when scanning through the Markdown. Having a dedicated bullet list for some example/reference components would be nice, so that people looking just for links to examples don't have to read everything to find the links.

File: docs/developer/writing-exporter-flow-components.md (hunk 1 above)
Comment (rfratto) [reply docs/developer/writing-exporter-flow-components.md:44]: For the second point, new components must be listed here: https://github.com/grafana/agent/blob/main/component/all/all.go

Comment (marctc) [reply docs/developer/writing-exporter-flow-components.md:44]: Ah right, adding now!

File: docs/developer/writing-exporter-flow-components.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
+# Create Prometheus Exporter Flow Components
+
+This guide will walk you through the process of creating a new Prometheus exporter Flow component and best practices for implementing it. It is required that the exporter has an existing integration in order to wrap it as a Flow component.
+
+## Arguments (Configuration)
+
+`Arguments` struct defines the arguments that can be passed to the component. In most cases, this would be exactly the same as the arguments that the integration for this exporter the uses. Some recommendations:
```
[the remark this answers is not in the document]
Comment (marctc) [reply docs/developer/writing-exporter-flow-components.md:14]: The full sentence is: "Avoid creating components with `Singleton: true` as it will make it impossible to run multiple instances of the exporter." would be that enough?

**Review by jcreixell — APPROVED** [review]
:heart_eyes: extra kudos for mentioning adding tests :see_no_evil:

File: docs/developer/writing-exporter-flow-components.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
+# Create Prometheus Exporter Flow Components
+
+This guide will walk you through the process of creating a new Prometheus exporter Flow component and best practices for implementing it. 
+
+It is required that the exporter has an existing integration in order to wrap it as a Flow component. In the future, we will drop this requirement and Flow components will expose the logic of the exporter directly.
```
Comment (gloriasc) [inline docs/developer/writing-exporter-flow-components.md:5]: What does this mean "Have an existing integration" do you mean an existing cloud integration? The word is so overused that I can't tell what it means here.

File: docs/developer/writing-exporter-flow-components.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+Use the following exporters as a reference:
+- [process_exporter](../../component/prometheus/exporter/process/process.go) - [documentation](../sources/flow/reference/components/prometheus.exporter.process.md)
+- [blackbox_exporter](../../component/prometheus/exporter/blackbox/blackbox.go) - [documentation](../sources/flow/reference/components/prometheus.exporter.blackbox.md)
+- [node_exporter](../../component/prometheus/exporter/unix/unix.go) - [documentation](../sources/flow/reference/components/prometheus.exporter.unix.md)
+
+## Arguments (Configuration)
+
+`Arguments` struct defines the arguments that can be passed to the component. In most cases, this would be exactly the same as the arguments that the integration for this exporter the uses. Some recommendations:
```
Comment (gloriasc) [inline docs/developer/writing-exporter-flow-components.md:14]: Nitpick: There is an superfluous  "the" 

> this would be exactly the same as the arguments that the integration for this exporter **the** uses.

File: docs/developer/writing-exporter-flow-components.md (hunk 6 above)
Comment (marctc) [reply docs/developer/writing-exporter-flow-components.md:5]: Great point. I'm talking about agent's integrations. I'd reword it.
